pkg/operation/botanist/controlplane.go

```diff
@@ -73,8 +118,91 @@ func (b *Botanist) DeleteKubeAPIServer() error {
 	return err
 }
 
+// MoveTerraformResources copies the terraform resources realted to backup infrastructure creation from  a shoot's main namespace
+// in the Seed cluster to Shoot's backup namespace.
+func (b *Botanist) MoveTerraformResources() error {
+	t := terraformer.NewFromOperation(b.Operation, common.TerraformerPurposeBackup)
+	// Clean up possible existing job/pod artifacts from previous runs
+	jobPodList, err := t.ListJobPods()
+	if err != nil {
+		return err
+	}
+	if err := t.CleanupJob(jobPodList); err != nil {
+		return err
+	}
+	if err := t.WaitForCleanEnvironment(); err != nil {
+		return err
+	}
+	backupInfrastructureName := common.GenerateBackupInfrastructureName(b.Shoot.SeedNamespace, b.Shoot.Info.Status.UID)
+	backupNamespace := common.GenerateBackupNamespaceName(backupInfrastructureName)
+	oldPrefix := fmt.Sprintf("%s.%s", b.Shoot.Info.Name, common.TerraformerPurposeBackup)
+	newPrefix := fmt.Sprintf("%s.%s", backupInfrastructureName, common.TerraformerPurposeBackup)
+	tfConfig, err := b.K8sSeedClient.GetConfigMap(b.Shoot.SeedNamespace, oldPrefix+common.TerraformerConfigSuffix)
+	if err == nil {
+		if _, err := b.K8sSeedClient.CreateConfigMap(backupNamespace, newPrefix+common.TerraformerConfigSuffix, tfConfig.Data, true); err != nil {
+			return err
+		}
+		if err = b.K8sSeedClient.DeleteConfigMap(b.Shoot.SeedNamespace, oldPrefix+common.TerraformerConfigSuffix); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	} else if !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	tfState, err := b.K8sSeedClient.GetConfigMap(b.Shoot.SeedNamespace, oldPrefix+common.TerraformerStateSuffix)
+	if err == nil {
+		if _, err := b.K8sSeedClient.CreateConfigMap(backupNamespace, newPrefix+common.TerraformerStateSuffix, tfState.Data, true); err != nil {
+			return err
+		}
+		if err = b.K8sSeedClient.DeleteConfigMap(b.Shoot.SeedNamespace, oldPrefix+common.TerraformerStateSuffix); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	} else if !apierrors.IsNotFound(err) {
+		return err
+	}
+	tfVariables, err := b.K8sSeedClient.GetSecret(b.Shoot.SeedNamespace, oldPrefix+common.TerraformerVariablesSuffix)
+	if err == nil {
+		if _, err := b.K8sSeedClient.CreateSecret(backupNamespace, newPrefix+common.TerraformerVariablesSuffix, corev1.SecretTypeOpaque, tfVariables.Data, true); err != nil {
+			return err
+		}
+		if err = b.K8sSeedClient.DeleteSecret(b.Shoot.SeedNamespace, oldPrefix+common.TerraformerVariablesSuffix); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	} else if !apierrors.IsNotFound(err) {
+		return err
+	}
+	//TODO etcd-backup secret, no need just confirm
+	//TODO what about jobs and pods??
```

Comment:
what do u mean? you have already introduced the "wait for clean environment" logic in the beginning of the function, should be fine?!

Reply:
Yes will cleanup the comments. Sorry i cleaned it up earlier but i think i mess up finally while reverting some changes